### PR TITLE
feat(ucs): add new resource cluster

### PR DESCRIPTION
**Pull request by Jason-Zhang9309**
**What this PR does / why we need it**:

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note

```

## PR Checklist

* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud/services/acceptance/ucs' TESTARGS='-run TestAccCluster_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/ucs -v -run TestAccCluster_basic -timeout 360m -parallel 4
=== RUN   TestAccCluster_basic
=== PAUSE TestAccCluster_basic
=== CONT  TestAccCluster_basic
--- PASS: TestAccCluster_basic (591.65s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/ucs       591.722s

make testacc TEST='./huaweicloud/services/acceptance/ucs' TESTARGS='-run TestAccCluster_cce'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/ucs -v -run TestAccCluster_cce -timeout 360m -parallel 4
=== RUN   TestAccCluster_cce
=== PAUSE TestAccCluster_cce
=== CONT  TestAccCluster_cce
--- PASS: TestAccCluster_cce (534.03s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/ucs       534.098s
```

### CODE REVIEW

File: docs/resources/ucs_cluster.md [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+```
+
+Note that the imported state may not be identical to your resource definition, due to some attributes missing from the
+API response, security or some other reason. The missing attributes include: `annotations`.
+It is generally recommended running `terraform plan` after importing a cluster.
+You can then decide if changes should be applied to the cluster, or the resource definition
+should be updated to align with the cluster. Also you can ignore changes as below.
+
+```
+resource "huaweicloud_dws_cluster" "test" {
```
Comment (houpeng80) [inline docs/resources/ucs_cluster.md]:
```suggestion
resource "huaweicloud_ucs_cluster" "test" {
```

File: huaweicloud/services/ucs/resource_huaweicloud_ucs_cluster.go [hunk 2]
```diff
@@ -0,0 +1,402 @@
...
+			}
+			_, err = updateClusterClient.Request("POST", removeClusterFormFleetPath, &removeClusterFormFleetOpt)
+			if err != nil {
+				return diag.Errorf("error updating Cluster: %s", err)
+			}
+		}
+
+		if newFleetID.(string) != "" {
+			var (
+				removeClusterFormFleetHttpUrl = "v1/clusters/{id}/join"
```
Comment (houpeng80) [inline huaweicloud/services/ucs/resource_huaweicloud_ucs_cluster.go]:
```suggestion
				addClusterToFleetHttpUrl = "v1/clusters/{id}/join"
```

File: docs/resources/ucs_cluster.md (hunk 1 above)
Comment (Jason-Zhang9309) [reply docs/resources/ucs_cluster.md]: fixed

File: huaweicloud/services/ucs/resource_huaweicloud_ucs_cluster.go (hunk 2 above)
Comment (Jason-Zhang9309) [reply huaweicloud/services/ucs/resource_huaweicloud_ucs_cluster.go]: fixed

**Pull request merged**